Src/FluentAssertions/DataRowCollectionAssertionExtensions.cs

```diff
@@ -0,0 +1,827 @@
+                {
+                    var context = new EquivalencyValidationContext(Node.From<DataRow>(CallerIdentifier.DetermineCallerIdentity), options)
+                    {
+                        Reason = new Reason(because, becauseArgs),
+                        TraceWriter = options.TraceWriter
+                    };
+
+                    var comparands = new Comparands
+                    {
+                        Subject = actualItem,
+                        Expectation = unexpected,
+                        CompileTimeType = typeof(DataRow),
+                    };
+
+                    new EquivalencyValidator().AssertEquality(comparands, context);
+
+                    string[] failures = scope.Discard();
+
+                    if (!failures.Any())
+                    {
+                        foundIndices.Add(index);
+                    }
+
+                    index++;
+                }
+            }
+
+            if (foundIndices.Count > 0)
+            {
+                using (new AssertionScope())
+                {
+                    Execute.Assertion
+                        .BecauseOf(because, becauseArgs)
+                        .WithExpectation("Expected {context:collection} {0} not to contain equivalent of {1}{reason}, ", assertion.Subject, unexpected)
+                        .AddReportable("configuration", options.ToString());
+
+                    if (foundIndices.Count == 1)
+                    {
+                        Execute.Assertion
+                            .FailWith("but found one at index {0}.", foundIndices[0]);
+                    }
+                    else
+                    {
+                        Execute.Assertion
+                            .FailWith("but found several at indices {0}.", foundIndices);
+                    }
+                }
+            }
+
+            return new AndConstraint<GenericCollectionAssertions<DataRow>>(assertion);
+        }
+
+        private static bool RowsAreEquivalent(AssertionScope scope, EquivalencyAssertionOptions<DataRow> options, DataRow actualItem, DataRow expectedItem, string because, object[] becauseArgs)
+        {
+            var context = new EquivalencyValidationContext(Node.From<DataRow>(() => CallerIdentifier.DetermineCallerIdentity()), options)
+            {
+                Reason = new Reason(because, becauseArgs),
+                TraceWriter = options.TraceWriter
+            };
+
+            var comparands = new Comparands
+            {
+                Subject = actualItem,
+                Expectation = expectedItem,
+                CompileTimeType = typeof(DataRow),
+            };
+
+            new EquivalencyValidator().AssertEquality(comparands, context);
+
+            string[] failures = scope.Discard();
+
+            return !failures.Any();
+        }
+
+        /// <summary>
+        /// Asserts that the <see cref="DataRowCollection"/> is a subset of the <paramref name="expectedSuperset" />.
```

Comment:
🤔 This documentation is insufficient to understand it. From what I see in the implementation, it's more like `ContainsEquivalentSubsetOf` because the normal `BeSubsetOf` for collections relies on its `Equals` implementation. The same applies to `NotBeSubsetOf` and `(Not)IntersectWith`.

Reply:
To me this does not fall under the PR title "Restore basic assertions for collections in System.Data" as we have never had assertions that combined subset/superset/intersection with equivalency comparisons.
I'd like to at least leave those out of this PR to limit the size of it.

Reply:
We can discuss this more properly in its own PR, but just to note, `DataRow` objects _don't really have_ any other type of equivalency, because they are tied to the `DataTable` that created them. Literally the only way to have a subset or intersection with collections of `DataRow`s is through equivalency.

Reply:
To present you (potentially) wasting work, please open an issue first where can have the initial discussions.